File: Children-Competition-Java/mpp-proiect-java-ABNegrea/Problema5/src/main/java/Concurs/Service/ParticipationService.java
package Concurs.Service;

import Concurs.Domain.Participation;
import Concurs.Repository.ParticipationDBRepository;

import java.util.List;
import java.util.UUID;

public class ParticipationService {
    private final ParticipationDBRepository participationRepo;

    public ParticipationService(ParticipationDBRepository participationRepo) {
        this.participationRepo = participationRepo;
    }
    public Participation addParticipation(Participation participation){
        return this.participationRepo.save(participation);
    }

    public List<Participation> findAll()
    {
        return participationRepo.findAll().stream().toList();
    }

    public int participationCountChild(UUID id){
        return this.participationRepo.participationCountChild(id);
    }
}

File: Children-Competition-C#/Service/ParticipationService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Concurs.Domain;
using Concurs.Repository;

namespace Concurs.Service
{
    internal class ParticipationService
    {
        private ParticipationDBRepository participationDBRepository;

        public ParticipationService(ParticipationDBRepository participationDBRepository)
        {
            this.participationDBRepository = participationDBRepository;
        }

        public Participation AddParticipation(Participation participation)
        {
            return this.participationDBRepository.Save(participation);
        }
        public List<Participation> FindAll()
        {
            return (List<Participation>)this.participationDBRepository.FindAll();
        }
        public int ParticipationCountChild(Guid IdChild)
        {
            return participationDBRepository.ParticipationCountChild(IdChild);
        }
    }
}
